### PR TITLE
Improved: Converted the setPaymentStatus service from XML to groovy DSL. [OFBIZ-11482]

### PR DESCRIPTION
**Pull request by vyasdevanshu**

### CODE REVIEW

File: applications/accounting/groovyScripts/payment/PaymentServices.groovy [hunk 1]
```diff
@@ -94,3 +94,66 @@ def getPaymentRunningTotal(){
...
     return result
 }
+
+def setPaymentStatus() {
+    if (parameters.paymentId) {
+        Map resultMap = new HashMap()
+        payment = from("Payment").where("paymentId", parameters.paymentId).queryOne()
+        if (payment) {
+            oldStatusId = payment.statusId
+            statusItem = from("StatusItem").where("statusId", parameters.statusId).queryOne()
```
Comment (dixitdeepak) [inline applications/accounting/groovyScripts/payment/PaymentServices.groovy:104]: We should use cache here.

File: applications/accounting/groovyScripts/payment/PaymentServices.groovy [hunk 2]
```diff
@@ -94,3 +94,66 @@ def getPaymentRunningTotal(){
...
+            if (statusItem) {
+                if (!oldStatusId.equals(parameters.statusId)) {
+                    statusChange = from("StatusValidChange").where("statusId", oldStatusId, "statusIdTo", parameters.statusId).queryOne()
```
Comment (dixitdeepak) [inline applications/accounting/groovyScripts/payment/PaymentServices.groovy:107]: Use cache.

File: applications/accounting/groovyScripts/payment/PaymentServices.groovy [hunk 3]
```diff
@@ -94,3 +94,66 @@ def getPaymentRunningTotal(){
...
+                        if (("PMNT_RECEIVED".equals(parameters.paymentId) || "PMNT_SENT".equals(parameters.paymentId)) && !payment.paymentMethodId) {
+                            resultMap.errorMessage = UtilProperties.getMessage("AccountingUiLabels","AccountingMissingPaymentMethod", [statusItem.description], locale)
+                            logError("Cannot set status to " + parameters.statusId + " on payment " + payment.paymentId + ": payment method is missing")
+                            return failure(resultMap.errorMessage)
+                        }
+                        // check if the payment fully applied when set to confirmed
+                        if ("PMNT_CONFIRMED".equals(parameters.statusId)) {
+                            notYetApplied = org.apache.ofbiz.accounting.payment.PaymentWorker.getPaymentNotApplied(payment)
+                            if (BigDecimal.ZERO.compareTo(notYetApplied)) {
+                                resultMap.errorMessage = UtilProperties.getMessage("AccountingUiLabels","AccountingPSNotConfirmedNotFullyApplied", locale)
```
Comment (dixitdeepak) [inline applications/accounting/groovyScripts/payment/PaymentServices.groovy:119]: I don't think we need to assign this to map, resultMap is unused here

File: applications/accounting/groovyScripts/payment/PaymentServices.groovy [hunk 4]
```diff
@@ -94,3 +94,66 @@ def getPaymentRunningTotal(){
...
+                                updateOrderPaymentPreferenceMap.statusId = "PAYMENT_CANCELLED"
+                                runService('updateOrderPaymentPreference', updateOrderPaymentPreferenceMap)
+                            }
+                        }
+
+                        // everything ok, so now change the status field
+                        payment.statusId = parameters.statusId
+                        payment.store()
+                    } else {
+                        resultMap.errorMessage = "Cannot change from " + oldStatusId + " to " + parameters.statusId
```
Comment (dixitdeepak) [inline applications/accounting/groovyScripts/payment/PaymentServices.groovy:146]: I think we should use CommonErrorNoStatusValidChange uiLabel here.

**Comment by nmalin** [comment]
Hello, I merged on trunk this service migration based on your patch @vyasdevanshu  and @dixitdeepak 's remarks

**Pull request closed without merging**